apps/services/notifications/src/app/modules/notifications/dto/createNotification.dto.ts

```diff
@@ -0,0 +1,36 @@
+import { IsEnum, IsNotEmpty, IsString } from 'class-validator'
+import { ApiProperty } from '@nestjs/swagger'
+import { IsNationalId } from '@island.is/nest/validators'
+
+export enum MessageTypes {
+  NewPostholfMessage = 'newPostholfMessage',
+}
+
+export class TypeValidator {
+  @IsEnum(MessageTypes)
+  @ApiProperty({ enum: MessageTypes })
+  type!: MessageTypes
+}
+
+export class NewPostholfMessage {
+  @ApiProperty({ enum: [MessageTypes.NewPostholfMessage] })
+  type!: MessageTypes.NewPostholfMessage
+
+  @IsString()
```

Comment:
Should we maybe validate that this string is of a valid date format e.g. with `@isDateString`

Reply:
This is actually the name of the institution sending the message.

Maybe it would be clearer to name this field `sender`

Reply:
Haha, yes that might be a bit clearer